virttest/utils_misc.py

```diff
@@ -4429,3 +4429,21 @@ def _md5(fd):
         if _md5(fd_a) == _md5(fd_b):
             return True
     return False
+
+
+def get_real_file_path(file_path):
+    """
+    Check if the file is link file, if yes, get the real
+    file path
+
+    param file_path: the file's path
+    """
+    LOG.info("Get %s real file path" % file_path)
+    if os.path.islink(file_path):
+        real_path = os.path.realpath(file_path)
+        if os.path.islink(real_path):
+            get_real_file_path(real_path)
+        else:
+            return real_path
+    else:
+        return file_path
```

Comment:
```suggestion
def get_real_path(org_path):
    """
    Check if the file/directory is a real path, get the real
    path if it is not

    param org_path: the orginial path
    """

    LOG.debug("Get real path of %s" % org_path)
    if os.path.islink(org_path):
        real_path = os.path.realpath(org_path)
        return get_real_path(real_path)
    else:
        return org_path
```

Reply:
@leidwang Ah, `os.path.realpath` is recursive already, I suppose this function was not needed at all then. Would you please elaborate on the actual scenario which I probably omitted?

Reply:
Yes, you are right.Thanks for the guidance.

https://docs.python.org/3/library/os.path.html#os.path.realpath